pennylane/operation.py

```diff
@@ -959,6 +959,30 @@ def compute_decomposition(*params, wires=None, **hyperparameters):
         """
         raise DecompositionUndefinedError
 
+    # pylint: disable=no-self-argument, comparison-with-callable
+    @classproperty
+    def has_decomposition_threshold(cls):
```

Comment:
We've started adding `has_` properties for the cases where:
a) sometimes we just want to know the existence of the property (like matrix)
b) it is expensive to compute by itself (like matrix)

I'd say it's fairly cheap to check `op.decomposition_threshold is not None`, so case b does not hold.